### PR TITLE
bugfix: change column name and type to json

### PR DESCRIPTION
**Pull request by olyazavr**
## Description
When trying to change the type and name of a column to a JSON type, the onlineddl migration fails because it cannot find the new column name in the old table. This specifically only fails when converting to JSON, not any other type.

This is because it is parsing the `convert` and using the alias (which is a column that does not exist in the table) as the column it will select from. This is a problem in two places.

See https://github.com/vitessio/vitess/issues/14089 for repro steps

## Related Issue(s)

Fixes https://github.com/vitessio/vitess/issues/14089

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by shlomi-noach** [comment]
The solution looks correct. Let me push a `onlineddl_vrepl_suite` test on top, and then do a proper review.

**Comment by shlomi-noach** [comment]
Huh. I'm unable to push into this PR's branch (I should be able to, this was a nice touch GitHub added where maintainers could push to a remote branch offered as a PR).

**Comment by shlomi-noach** [comment]
Resubmitted as https://github.com/vitessio/vitess/pull/14096

File: go/vt/vttablet/tabletmanager/vreplication/table_plan_builder.go [hunk 1]
```diff
@@ -425,9 +425,23 @@ func (tpb *tablePlanBuilder) analyzeExpr(selExpr sqlparser.SelectExpr) (*colExpr
 		references: make(map[string]bool),
 	}
 	if expr, ok := aliased.Expr.(*sqlparser.ConvertUsingExpr); ok {
+		var colName sqlparser.IdentifierCI
```
Comment (shlomi-noach) [inline go/vt/vttablet/tabletmanager/vreplication/table_plan_builder.go:428]: Please add a comment here (a short paragraph) explaining why the following code is necessary.

File: go/vt/vttablet/tabletserver/vstreamer/planbuilder.go [hunk 2]
```diff
@@ -727,7 +727,21 @@ func (plan *Plan) analyzeExpr(vschema *localVSchema, selExpr sqlparser.SelectExp
 			FixedValue: sqltypes.NewInt64(num),
 		}, nil
 	case *sqlparser.ConvertUsingExpr:
-		colnum, err := findColumn(plan.Table, aliased.As)
+		var colName sqlparser.IdentifierCI
```
Comment (shlomi-noach) [inline go/vt/vttablet/tabletserver/vstreamer/planbuilder.go:730]: Similarly please add a comment here as well.

**Review by shlomi-noach — APPROVED** [review]
Thank you for the PR! This looks good, and appreciate the tests.

Can you please merge `planetscale:fix-onlineddl-change-col-json` (https://github.com/vitessio/vitess/pull/14096) into your branch? It adds a `onlineddl_vrepl_suite` test that validates the complete cycle.

Will backport to supported versions.

**Comment by olyazavr** [comment]
Comments added and tests pulled in, thanks for the quick response!

**Review by mattlord — APPROVED** [review]
LGTM! Thank you @olyazavr ! ❤️ 

@manan and @harshit-gangal would either of you mind taking a quick look as well, and help with the backports in the lead up to our release work tomorrow? 🙏

**Pull request merged**